### DISCUSSION TITLE
Tell click to stop parsing after specific point

### DISCUSSION THREAD

**Discussion by link89** [pallets/click#2354] [Q&A]
**Feature Description** 

I am working on a tool to wrap some existed command line tools to make them easier to use (like auto install, inject argument automatically, etc). And I hope that I can have a way to tell `click` to stop parsing after encount some specific command.

**Example**
Here are some real world examples like `poetry run python3 ./my-scripts.py`  or `conda run -n my-env python3 ./my-scripts.py`

**Comment by jules43** [comment]
I'm not absolutely clear on what you're trying to achieve. Are you looking to invoke an existing app that uses click for command line processing or to modify/write an app that uses click to process the command line?

You mention stopping parsing. Click option processing supports a way to mess with processing but I'm not sure it is relevant to your problem:

https://click.palletsprojects.com/en/8.1.x/options/#callbacks-and-eager-options

**Comment by davidism** [comment]
Add an argument with `nargs=-1`, it will collect any tokens. Use `--` if those tokens might look like options:

```python
@cli.command()
@click.argument("parts", nargs=-1)
def run(parts):
    ...
```

```
mycli run -- python -m http.server
```

**Reply by link89** [reply]
Yes, that's the idea, actually I have create a PR for another cmd framework `fire` to support this feature: https://github.com/google/python-fire/issues/403
